pkg/cloudprovider/provider/openstack/provider.go

```diff
@@ -316,10 +290,6 @@ func (p *provider) Validate(spec v1alpha1.MachineSpec) error {
 		return fmt.Errorf("failed to get network %q: %v", c.Network, err)
 	}
 
-	if _, err := getSubnet(client, c.Region, c.Subnet); err != nil {
```

Comment:
Could you replace this with a check that checks if the configured network as at least one subnet?

Reply:
ok, I can do that

Reply:
Another thought is we need defer those kind of logic into provider... 

I created a net without subnet and deploy will have following error: I tried in machine controller and didn't see error, however 

stack@k8s:~/devstack$ openstack server create --key-name k1 --image 55904f0b-ff9e-4b4c-91d5-0b98a4713e32 --flavor 1 --nic net-id=ad8b1a43-27eb-498a-b0db-5ce271e1050f --security-group group1 jji1
Network ad8b1a43-27eb-498a-b0db-5ce271e1050f requires a subnet in order to boot instances on. (HTTP 400) (Request-ID: req-c0c15150-5c11-4166-b0a1-29e0ffe71ca5)


  Type     Reason                Age                From                Message
  ----     ------                ----               ----                -------
  Warning  CreateInstanceFailed  47s (x3 over 67s)  machine-controller  Instance creation failed: failed to create server, due to Invalid request due to incorrect syntax or missing required parameters.
  Warning  CreateError           47s (x3 over 67s)  machine-controller  failed to create server, due to Invalid request due to incorrect syntax or missing required parameters.. Unable to create a machine.
root@kvm-017407:~/go/src/github.com/kubermatic/machine-controller# kubectl get machine -n kube-system
NAME                                           PROVIDER    OS       ADDRESS
openstack-machinedeployment-5df56b457c-stslv   openstack   ubuntu


so my thinking is
1) let's defer the creation error detect for complicate logics, this kind of logic check will be hard to maintain in k8s layer, use openstack and get its output then handle it will make sense (grammar and whether something is existing check is another story) 

2) when error occurs, get machine should output error info...

root@kvm-017407:~/go/src/github.com/kubermatic/machine-controller# kubectl get machine -n kube-system
NAME                                           PROVIDER    OS       ADDRESS
openstack-machinedeployment-5df56b457c-stslv   openstack   ubuntu

Reply:
ok, I figured it out, need update CRD xml file...